MM-42267: Update browse channel UI

sinansonmez opened the pull request:
#### Summary
UI update for browse channel model. 

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/21173
JIRA: https://mattermost.atlassian.net/browse/MM-42267

#### Screenshots


|  before  |  after  |
|----|----|
| <img width="669" alt="Screen Shot 2023-05-29 at 18 01 29" src="https://github.com/mattermost/mattermost-server/assets/37421564/cea35591-dda2-4b31-baa3-692297bfb04c"> | <img width="669" alt="Screen Shot 2023-05-29 at 16 43 00" src="https://github.com/mattermost/mattermost-server/assets/37421564/83635bbc-2868-4cf0-9796-f6e6be14b129"> |


#### Release Note
```release-note
Updated UI for browse channel modal
```

mattermost-build commented:
Hello @sinansonmez,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

mattermost-build commented:
E2E tests not automatically triggered, because PR has no approval yet. Please ask a developer to review and then try again to attach the QA label.

sinansonmez commented:
As agreed with @hmhealey I have removed the member count from the modal as the current API to get info for each channel is quite heavy task. 

@matthewbirtch One question. As we have removed the member count, do we still need to show a small dot before channel description in the channels that we are not member (as an example, ut or veritatis channels in the before screenshot)?

esethna commented:
Thanks @sinansonmez! Agree with removing the dot if there's nothing infront of it

matthewbirtch commented:
> As agreed with @hmhealey I have removed the member count from the modal as the current API to get info for each channel is quite heavy task.
> 
> @matthewbirtch One question. As we have removed the member count, do we still need to show a small dot before channel description in the channels that we are not member (as an example, ut or veritatis channels in the before screenshot)?

Yep, makes sense to remove the dot If we're no longer showing the member count. I wasn't aware that we needed to do that, but if there are performance implications then so be it.

sinansonmez commented:
/update-branch

sinansonmez commented:
> > As agreed with @hmhealey I have removed the member count from the modal as the current API to get info for each channel is quite heavy task.
> > @matthewbirtch One question. As we have removed the member count, do we still need to show a small dot before channel description in the channels that we are not member (as an example, ut or veritatis channels in the before screenshot)?
> 
> Yep, makes sense to remove the dot in this case if we're no longer showing the member count. I wasn't aware that we needed to do that, but if there are performance implications then so be it.

@matthewbirtch @esethna dot is removed if there is nothing before it

<img width="669" alt="image" src="https://github.com/mattermost/mattermost-server/assets/37421564/f612fb67-f099-4c68-a7e1-2a30adfa423d">

sinansonmez commented:
/e2e-tests

mattermost-build commented:
You don't have permissions to trigger this command.
 It's only available for organization members.

sinansonmez commented:
@esethna can you request review from developers? As there were some internal changes in MM, I dont who should review this?

sinansonmez commented:
@hmhealey any idea?

hmhealey commented:
The build hash in the About Mattermost modal is `3b0d0d7179590e467e3225daddcb876e30ea8bf2` which is from master branch earlier today. I'll ask around to see if anyone knows why that might have happened. It's possible that it's because we finally changed the repo's name

hmhealey commented:
@sinansonmez If you wouldn't mind resolving those conflicts and updating the branch to the latest master, that should be resolved. It was probably caused by the repo rename, but that should be fixed now

mattermost-build commented:
Error trying to update the PR.
Please do it manually.

sinansonmez commented:
@hmhealey I have updated it to latest master but test server is still not picking up the changes from the PR

hmhealey commented:
That's very strange. I tried submitting my own branch with the same changes in case the problem was related to you being on a fork, but its test server is on the same commit. I've tried pushing an empty to see if that changes anything, and I'll also reach out to see if the team who had fixed that issue earlier in the week knows why it's still stuck on master

saturninoabril commented on e2e-tests/cypress/tests/integration/channels/enterprise/accessibility/accessibility_modals_dialogs_spec.js:
```diff
@@ -104,30 +104,26 @@ describe('Verify Accessibility Support in Modals & Dialogs', () => {
-                    // # Focus on the Create Channel button and TAB twice
-                    cy.get('#createNewChannel').focus().tab().tab();
+                    // # Focus on the Create Channel button and TAB three time
+                     cy.get('#createNewChannelButton').focus().tab().tab().tab();
```
```suggestion
                    cy.get('#createNewChannelButton').focus().tab().tab().tab();
```

saturninoabril commented:
/upgrade-test

saturninoabril commented:
/upgrade-branch

saturninoabril commented:
/update-branch

saturninoabril commented:
/e2e-test

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/343876) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/343876)

sinansonmez commented:
thanks for running e2e tests @saturninoabril out of 3 failing tests I think 2 of them are related to my changes. I fixed them. Could you please run the tests again?
/update-branch

sinansonmez commented:
@hmhealey test server is working correctly. @matthewbirtch please take a look

matthewbirtch reviewed (approved):
Lookin' good @sinansonmez! Thank you!

M-ZubairAhmed commented:
/update-branch

M-ZubairAhmed commented:
/e2e-test

M-ZubairAhmed commented on webapp/channels/src/components/more_channels/index.ts:
0/5 but shouldn't this essentially called `browser_channels_modal` cc @hmhealey I know that would create a big diff, but being clear in naming is always a good fallback

sinansonmez replied:
0/5 to me as well

hmhealey replied:
Let's leave it as-is for now since this PR's pretty big. We can do a followup once this is in though to rename it to `BrowseChannelsModal` after though since that's a good point that it's not super clear right now.

Historically, the modal was titled More Channels, but that's obviously changed since then. It's also less clear than some of our other historical names like the System Console being `AdminConsole` which is less ambiguous than `MoreChannels`

hmhealey replied:
Oh, the "More Channels" name was just the current name of the modal. I didn't realize it stayed that way even when we renamed the menu item to "Browse Channels". I guess I had just assumed they matched, and now I feel dumb for explaining the current state of it like it was some ancient MM fact 😛

M-ZubairAhmed replied:
Awesome make sense @sinansonmez would you mind opening a follow up PR after we merge this on on the name change ?

sinansonmez replied:
Yeah sure

sinansonmez commented on webapp/channels/src/components/common/svg_images_components/magnifying_glass_svg.tsx:
@M-ZubairAhmed it is this image. Therefore I dont think it is possible to import this from compass-icons.
<img width="166" alt="Screenshot 2023-06-13 at 22 27 40" src="https://github.com/mattermost/mattermost/assets/37421564/a9736599-6c7c-4639-9794-15394f6e03ff">

M-ZubairAhmed replied:
Ah yes I see yes you are right

hmhealey commented on webapp/channels/src/components/more_channels/more_channels.tsx:
```diff
@@ -214,87 +250,87 @@ export default class MoreChannels extends React.PureComponent<Props, State> {
+
         return (
-            <Modal
-                dialogClassName='a11y__modal more-modal more-modal--action'
```
I'm glad to see that we're getting rid of the original usage of `more-modal`. That's one of those overused classes in MM that makes working with our CSS styling difficult, so this'll definitely help to get rid of that eventually

M-ZubairAhmed commented on webapp/channels/src/components/searchable_channel_list.tsx:
```diff
@@ -0,0 +1,481 @@
+    }
+}
+
+// SearchableChannelList.defaultProps = {
```
I really hoped since this was a new component we could have written this in functional way but it's fine either ways. Also do you plan to remove this ?

saturninoabril commented:
/e2e-test

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/344585) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/344585)

yasserfaraazkhan reviewed (approved):
🎉

sinansonmez commented:
@M-ZubairAhmed i think this is ready to merge
